packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -1270,6 +1298,27 @@ export class PipelineEditor extends React.Component<
       this.handleOpenPipeline();
     });
   }
+
+  componentDidUpdate(): void {
+    const inputFields = document.querySelectorAll('.properties-readonly');
```

Comment:
I wonder if there is a way to access how uihints are rendered from `properties.json` and use it instead of reading info from the DOM to create a tooltip from scratch?

Reply:
I agree - I tried looking into it and this was my last option :/ And it seemed better than having a long text field

I might be missing something, but from investigating I found that you could have descriptions based on the value of fields: https://github.com/elyra-ai/canvas/wiki/3.2-Common-Properties--UI-Hints#dynamic-text-expressions, but I wasn't able to get it working and it seems like it might be only for creating a sum or percent? 

I also looked into the `readonly` [code](https://github.com/elyra-ai/canvas/blob/04cc96417a35eb9aff5e409ab3398ef2ebbf77a9/canvas_modules/common-canvas/src/common-properties/controls/readonly/readonly.jsx#L98) and it seems like there's only a tooltip if the control is within a table? I don't know much about it and it doesn't seem to be documented anywhere.

Reply:
We can always clarify that with the canvas team later, I was just curious if we could reuse their tool, good to know you've already tried something out :) I'm fine with keeping it as it for now.